.github/workflows/swift-toolchain.yml

```diff
@@ -4116,6 +4151,47 @@ jobs:
             & "${{ github.workspace }}\SourceCache\mimalloc\bin\minject$BuildSuffix" -l "$Exe"
           }
 
+      - uses: azure/login@v2
```

Comment:
It might be worth it to move all of this to a testable reusable action. It's quite a bit of duplicated code and a test would be handy to figure out what is broken when things break.

Reply:
@compnerd How would you like to proceed? I think there is value in making this a separate re-usable sub-action:
* Easier maintenance burden when we need to update the code. It's very easy to miss a copy of the duplicated code.
* We can add a simple test for the re-usable action so when signing break, we can use that to test signing in isolation without needing to either create a new dummy workflow for testing purposes or running the whole build again.

WDYT?